src/divbuf.rs

```diff
@@ -388,6 +389,44 @@ impl From<Vec<u8>> for DivBufShared {
     }
 }
 
+impl TryFrom<DivBufShared> for Vec<u8> {
+    type Error = DivBufShared;
+
+    /// Attempt to extract the owned storage from a DivBufShared.
+    ///
+    /// This will fail if there are any other living DivBufs for this same
+    /// DivBufShared, in which case the DivBufShared will be returned
+    /// unmodified.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use divbuf::*;
+    /// use std::convert::TryInto;
+    /// let dbs = DivBufShared::from(vec![1, 2, 3, 4, 5, 6]);
+    /// let vec: Vec<u8> = dbs.try_into().unwrap();
+    /// assert_eq!(vec, [1, 2, 3, 4, 5, 6][..]);
+    /// ```
+    fn try_from(buf: DivBufShared) -> Result<Self, Self::Error> {
+        let inner = unsafe { &*buf.inner };
+        if inner.sharers.load(Relaxed) == 1 &&
```

Comment:
Relaxed is not the correct ordering.  I think it should be Release.

Reply:
Release isn't valid for loads, only Relaxed, Acquire, and SeqCst. I think we want Acquire.